vendor/bat-native-ads/src/bat/ads/internal/ml/pipeline/pipeline_util.cc

```diff
@@ -188,44 +188,38 @@ absl::optional<model::Linear> ParsePipelineClassifier(
 
 }  // namespace
 
-absl::optional<PipelineInfo> ParsePipelineJSON(std::string json) {
-  absl::optional<base::Value> root = base::JSONReader::Read(json);
-
-  // Free memory in advance to optimize the peak memory consumption. |json| can
-  // be up to 10Mb and the following code allocates an extra memory few Mb of
-  // memory.
-  json = std::string();
-
-  if (!root) {
+absl::optional<PipelineInfo> ParsePipelineValue(base::Value resource_value) {
```

Comment:
`resource_value` takes more than 100Mb and your solution passes it via threads (=it has to be stored between calls).
Have you measured peak the memory usage?

Reply:
Discussed in PM, need to do json parsing in background thread

Reply:
Can we please raise an issue for this change, thanks